src/daft-micropartition/src/partitioning.rs

```diff
@@ -0,0 +1,208 @@
+use std::{collections::HashMap, sync::Arc};
+
+use common_error::{DaftError, DaftResult};
+use daft_dsl::Expr;
+use futures::stream::BoxStream;
+
+use crate::MicroPartition;
+type PartId = String;
+
+#[derive(Debug, Clone)]
+pub struct Boundaries {
+    pub sort_by: Vec<Expr>,
+    pub bounds: Arc<MicroPartition>,
+}
+
+#[derive(Debug, Clone)]
+pub struct PartitionMetadata {
+    pub num_rows: usize,
+    pub size_bytes: usize,
+    pub boundaries: Option<Boundaries>,
+}
+
+impl PartitionMetadata {
+    pub fn from_micro_partition(mp: &MicroPartition) -> Self {
```

Comment:
nittttttttt (super nit): any reason this is not a `impl From`

Reply:
The naming here was copied over from the python implementation. I wanted to make as much a 1-1 as possible.

Reply:
ok sgtm